test/framework/provider.go

```diff
@@ -0,0 +1,95 @@
+/*
+Copyright 2021 The Crossplane Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package framework
```

Comment:
Might be a pre-mature optimization, but we could consider going ahead and moving this to `framework/provider/wait.go` in a `package provider`. That could allow for simpler function signatures, such as `provider.WaitForAllInstalled()`.